src/TurbulenceClosures/abstract_scalar_diffusivity_closure.jl

```diff
@@ -0,0 +1,147 @@
+
+Abstract type for closures with *isotropic* diffusivities.
+"""
+abstract type AbstractScalarDiffusivity{TD, Dir} <: AbstractTurbulenceClosure{TD} end
+
+"""
+    struct ThreeDimensional end
+
+Corresponds to a closure which is three-dimensionally isotropic
+"""
+struct ThreeDimensional end
+
+"""
+    struct Horizontal end
+
+Corresponds to a closure which is horizontally isotropic and calculated in a VectorInvariant form
+"""
+struct Horizontal end
+
+"""
+    struct Vertical end
+
+Corresponds to a closure which calculates only the vertical direction
+"""
+struct Vertical end
+
+"""
+    viscosity(closure, diffusivities)
+
+Returns the isotropic viscosity associated with `closure`.
+"""
+function viscosity end 
+
+"""
+    diffusivity(closure, diffusivities, tracer_indedx) where tracer_index
+
+Returns the isotropic diffusivity associated with `closure` and tracer index `c_idx`.
+"""
+function diffusivity end 
+
+#####
+##### Stress divergences
+#####
+
+const AID = AbstractScalarDiffusivity{<:Any, <:ThreeDimensional}
+const AHD = AbstractScalarDiffusivity{<:Any, <:Horizontal}
+const AVD = AbstractScalarDiffusivity{<:Any, <:Vertical}
+
+@inline viscous_flux_ux(i, j, k, grid, closure::AID, clock, U, args...) = - 2 * ν_σᶜᶜᶜ(i, j, k, grid, clock, viscosity(closure, args...), Σ₁₁, U.u, U.v, U.w)
+@inline viscous_flux_vx(i, j, k, grid, closure::AID, clock, U, args...) = - 2 * ν_σᶠᶠᶜ(i, j, k, grid, clock, viscosity(closure, args...), Σ₂₁, U.u, U.v, U.w)
+@inline viscous_flux_wx(i, j, k, grid, closure::AID, clock, U, args...) = - 2 * ν_σᶠᶜᶠ(i, j, k, grid, clock, viscosity(closure, args...), Σ₃₁, U.u, U.v, U.w)
+@inline viscous_flux_uy(i, j, k, grid, closure::AID, clock, U, args...) = - 2 * ν_σᶠᶠᶜ(i, j, k, grid, clock, viscosity(closure, args...), Σ₁₂, U.u, U.v, U.w)
+@inline viscous_flux_vy(i, j, k, grid, closure::AID, clock, U, args...) = - 2 * ν_σᶜᶜᶜ(i, j, k, grid, clock, viscosity(closure, args...), Σ₂₂, U.u, U.v, U.w)
+@inline viscous_flux_wy(i, j, k, grid, closure::AID, clock, U, args...) = - 2 * ν_σᶜᶠᶠ(i, j, k, grid, clock, viscosity(closure, args...), Σ₃₂, U.u, U.v, U.w)
+
+@inline viscous_flux_ux(i, j, k, grid, closure::AHD, clock, U, args...) = - ν_δᶜᶜᶜ(i, j, k, grid, clock, closure.ν, U.u, U.v)   
+@inline viscous_flux_vx(i, j, k, grid, closure::AHD, clock, U, args...) = - ν_ζᶠᶠᶜ(i, j, k, grid, clock, closure.ν, U.u, U.v)
+@inline viscous_flux_uy(i, j, k, grid, closure::AHD, clock, U, args...) = + ν_ζᶠᶠᶜ(i, j, k, grid, clock, closure.ν, U.u, U.v)   
+@inline viscous_flux_vy(i, j, k, grid, closure::AHD, clock, U, args...) = - ν_δᶜᶜᶜ(i, j, k, grid, clock, closure.ν, U.u, U.v)
+# are we sure about this?? This might not be consistent for AHD
+@inline viscous_flux_wx(i, j, k, grid, closure::AHD, clock, U, args...) = - ν_σᶠᶜᶠ(i, j, k, grid, clock, viscosity(closure, args...), ∂xᶠᶜᶠ, U.w)
+@inline viscous_flux_wy(i, j, k, grid, closure::AHD, clock, U, args...) = - ν_σᶜᶠᶠ(i, j, k, grid, clock, viscosity(closure, args...), ∂yᶜᶠᶠ, U.w)
```

Comment:
I guess this strikes the heart of what we mean by `Horizontal`...

What's the stress tensor for a "Horizontal" scalar diffusivity?

Reply:
To be consistent and symmetric (as vertical derivatives of horizontal velocities are not accounted for in the vertical stress of the horizontal momentum) we neglect the vertical shear in the horizontal stress of vertical momentum. 

This formulation might not conserve angular momentum, but if the grid is highly anisotropic it is not a problem (it would be a problem to include the additional terms)